### ISSUE TITLE
No Screens Found Error

### ISSUE THREAD

**Issue by kahverengiKod** [dukzcry/crap#3]
Hey, idk do you still continue support to this util / repo. But i have got some problems with this util.

I have got nvidia-dkms package, also i'm using Bumblebee project and Nvidia-Xrun project. And they both work good. But when i try to start any program it doesn't starts and it gives no screens found error on terminal. I tested with this commands: ./primerun.sh steam & ./primerun.sh glxgears

Also i have got Arch Linux, GTX960M and KDE. (i tested this commands on TTY & KDE no works still same error)

I was getting this error on Bumblebee & Nvidia-Xrun projects. But i fixed this problem with installing right GPU DKMS package. (if i install nvidia-xrun on my computer it uses nvidia package but Bumblebee works with nvidia-dkms. xrun install that package and not working but if i install nvidia-dkms package it works great)

**Comment by dukzcry** [comment]
Hi. Please run "primerun.sh glxgears" again and attach "primerun.sh glxgears" console output, dmesg output (only after your run primerun.sh glxgears not before) and /var/log/X.1.log (check primerun.sh output for correct X.log path and name)

**Comment by crzyBrown** [comment]
Of course, the console log: https://hastebin.com/upibijavic.sql
X.1.log: https://hastebin.com/inilusetus.sql

**Comment by dukzcry** [comment]
```
[ 37819.225] (II) Module "ramdac" already built-in
[ 37819.226] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
[ 37819.226] (EE) NVIDIA:     system's kernel log for additional error messages and
[ 37819.226] (EE) NVIDIA:     consult the NVIDIA README for details.
[ 37819.226] (EE) No devices detected.
```
Please also attach dmesg output after you run primerun.sh (at least nvidia related messages at the end)

**Comment by crzyBrown** [comment]
Here: https://hastebin.com/gunefodefi.vbs

**Comment by crzyBrown** [comment]
I think this is about: 

`[37819.107468] NVRM: API mismatch: the client has the version 390.67, but
               NVRM: this kernel module has the version 396.24.  Please
               NVRM: make sure that this kernel module and all NVIDIA driver
               NVRM: components have the same version.`

but i don't know how to fix.

**Comment by dukzcry** [comment]
Yes, exactly. Thanks. I will try to modify script to install correct module. Will let you know.

**Comment by crzyBrown** [comment]
Still working on it?

**Comment by crzyBrown** [comment]
Oh yeah sorry :)

**Comment by dukzcry** [comment]
Done, please test. I will reply on monday.

**Comment by crzyBrown** [comment]
No work; 
console: https://paste.ubuntu.com/p/qXFPM9KZFT/

/var/log/X.1.log: https://paste.ubuntu.com/p/8XFz8FWzNQ/

dmesg: https://paste.ubuntu.com/p/wB2DRCMQHX/

**Comment by dukzcry** [comment]
ls /lib/modules please

**Comment by crzyBrown** [comment]
[kahverengi@archbox ~]$ ls /lib/modules
4.14.49-1-lts  4.17.2-1-ARCH  extramodules-4.14-lts  extramodules-4.17-ARCH

**Comment by dukzcry** [comment]
That's the culprit. The builder in package assumes there is only one modules dir https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/nvidia-x11/builder.sh#L20
So for test time please move everything except 4.14.49-1-lts dir out of there. Of course you'll then will miss modules for all kernels except 4.14.49-1-lts you're currently booted in.

**Comment by dukzcry** [comment]
@crzyBrown to fix the last issue in proper way, I've created a PR in upstream repo https://github.com/NixOS/nixpkgs/pull/42516
Please support it in some way (like, comment, ...)

**Comment by dukzcry** [comment]
@crzyBrown Unfortunately not yet merged :(

**Comment by dukzcry** [comment]
@crzyBrown now yes (wasn't at the time I've checked). Now we need to wait once that commit go into appropriate channel. I believe its nixpkgs-unstable (https://github.com/NixOS/nixpkgs-channels/commits/nixpkgs-unstable), you can check its nixpkgs-unstable via `nix-channel --list` command

**Comment by dukzcry** [comment]
Ok, we may continue
1. `nix-channel --update`
2. Update this script to latest version https://raw.githubusercontent.com/dukzcry/crap/master/primerun/primerun.sh

**Comment by crzyBrown** [comment]
Nope, still not working with `No Screen Found`
The console: https://hastebin.com/nopidivodu.go

**Comment by dukzcry** [comment]
When building nvidia drivers you have error
```Building linux driver against kernel: 
make[1]: *** /lib/modules/4.14.49-1-lts/source: No such file or directory.  Stop.
make: *** [Makefile:79: modules] Error 2
```
Do you have sources of your current kernel laying anywhere? They are needed to build nvidia drivers. Put source dir or symlink it as `/lib/modules/4.14.49-1-lts/source`. Then repeat
Example
```
ls /lib/modules/4.16.17/source/
arch/  block/  build/  certs/  crypto/  Documentation/  fs/  include/  init/  ipc/  kernel/  lib/  Makefile  mm/  net/  samples/  scripts/  security/  sound/  tools/  virt
```

**Comment by crzyBrown** [comment]
I don't know, how i can do?

**Comment by dukzcry** [comment]
ls /usr/src please

**Comment by crzyBrown** [comment]
`[kahverengi@archbox ~]$ ls /usr/src`
`bbswitch-0.8  gmock  gtest  nvidia-396.24  vmware-workstation-14.1.2_8497320`

**Comment by dukzcry** [comment]
give me a bit to install Arch into vm, I don't know it as well :D

**Comment by dukzcry** [comment]
Tried in vm. `sudo pacman -S linux-lts-headers` fixed issue for me

**Comment by dukzcry** [comment]
Please try latest version. Built ok for me in VM
> Also when i install linux-lts-headers it removed my bbswitch, nvidia and vmware-workstation wtf?

You need to update Arch. You have kernel 4.14.49-1-lts but headers are 4.14.52-1. So pacman removed dkms modules, but was unable to build them again since you have version mismatch there. I think you could try just `sudo pacman -Sy linux-lts` then reboot.

**Comment by crzyBrown** [comment]
I upgraded everyting, still not working :/ No Screen Error still same.

**Comment by crzyBrown** [comment]
OH I'm sorry, i forget to update primerun.sh. I updated and tested i got another interesting error now: 
https://hastebin.com/yedimanife.sql

**Comment by dukzcry** [comment]
Maybe http://www.jaredlog.com/?p=115 ?
I will try Arch usb install now :D

**Comment by dukzcry** [comment]
Please update script again
And you'd better test with glxinfo and glxgears first

**Comment by crzyBrown** [comment]
I updated script and tested with glxinfo same result:
https://hastebin.com/oganavocor.scala

**Comment by dukzcry** [comment]
This is glxgears, also mistyped as glxgear

**Comment by dukzcry** [comment]
Plz update script again
Steam game on Arch https://youtu.be/mfS3HJjfnj0
GTA V on NixOS https://youtu.be/5ceVaSnDIbI

**Comment by crzyBrown** [comment]
Updated script and nothing on screen again. Just i can see some info about glxgears like this `234960 frames in 5.0 seconds = 46991.965 FPS`

One more thing i'm using on KDE on Arch so i started terminal on KDE and tested on KDE with terminal. Command is `./primerun.sh glxgears`

I don't see any error but i see the command works. Interesting

Also can i reach you in anywhere? Like Discord, Steam or anything else...
I have got a idea for this script if i can be able to run.

**Comment by dukzcry** [comment]
I have a suspision it just can't auto-switch console like you see in my video.
WM/DE shouldn't matter. On the video I run it from XFCE, the image is from https://sourceforge.net/projects/arch-linux-usb-os/
Do you see it if you press ctrl+alt+f1 after script run? Then to get back press ctrl+alt+f7 (KDE should be on f7, if not - try other from f1 to f12).
Steam http://steamcommunity.com/id/dukzcry
